raiden-ts/src/channels/reducer.ts

```diff
@@ -73,55 +73,34 @@ function channelOpenFailureReducer(
   return unset(path, state);
 }
 
-function channelDepositSuccessReducer(
+function channelUpdateOnchainBalanceStateReducer(
   state: RaidenState['channels'],
-  action: channelDeposit.success,
+  action: channelDeposit.success | channelWithdrawn,
 ): RaidenState['channels'] {
   const path = [action.meta.tokenNetwork, action.meta.partner];
   let channel: Channel | undefined = get(path, state);
   if (!channel || channel.state !== ChannelState.open || channel.id !== action.payload.id)
     return state;
-  if (action.payload.participant === action.meta.partner)
-    channel = {
-      ...channel,
-      partner: {
-        ...channel.partner,
-        deposit: action.payload.totalDeposit,
-      },
-    };
-  else
-    channel = {
-      ...channel,
-      own: {
-        ...channel.own,
-        deposit: action.payload.totalDeposit,
-      },
-    };
-  return set(path, channel, state);
-}
 
-function channelWithdrawReducer(
-  state: RaidenState['channels'],
-  action: channelWithdrawn,
-): RaidenState['channels'] {
-  const path = [action.meta.tokenNetwork, action.meta.partner];
-  let channel: Channel | undefined = get(path, state);
-  if (!channel || channel.state !== ChannelState.open || channel.id !== action.payload.id)
-    return state;
+  const key = channelWithdrawn.is(action) ? 'withdraw' : 'deposit';
```

Comment:
@andrevmatos is that OK or maybe should I explicitly check for each action and throw if not?

Reply:
One **never** throws in reducers. At most, if some condition isn't met, it returns unchanged `state`. But you can safely assume the union is strict there, don't worry.